.github/Pull_Request_template.md

```diff
@@ -1,9 +1,26 @@
-# Description
-Please include a summary of the change. For large or complex changes please include enough information to introduce your change and explain motivation for it.
+### Description 
+_Add a comprehensive description of proposed changes_
 
-Changes proposed in this pull request:
--
--
--
+Fixes # - _issue number(s) if exists_
+
+- [ ] I have reviewed my changes thoroughly before submitting this pull request.
+- [ ] I have commented my code, particularly in hard-to-understand areas.
+- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
+- [ ] The unit tests pass successfully.
+- [ ] I have run it locally and tested the changes extensively.
+- [ ] I have resolved any merge conflicts that might occur with the base branch.
+- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
+
+### Type of change
+
+_Choose one or multiple, leave empty if none of the other choices apply_
+
+_Add a respective label(s) to PR if you have permissions_
+
+- [ ] bug fix - _change that fixes an issue_
+- [ ] new feature - _change that adds functionality_
+- [ ] extension - _change that extends functionality_
+- [ ] tests - _change in tests_
+- [ ] infrastructure - _change in infrastructure and CI_
+- [ ] documentation - _documentation update_
```

Comment:
This can be reduced to one bullet point because labels are already explained in github PR interface:
```suggestion
- [ ] I added a respective label(s) to PR in case I have permissions for that
```